lib/recommendation-server.js

```diff
@@ -42,6 +43,12 @@ RecommendationServer.prototype = {
 
     this.events.subscribe('urlbar-change', this.search);
   },
+  maxCharsForRecommendation: function() {
+    const name = 'maxCharsForSearchSuggestions';
+    return this.urlbarPrefs.getPrefType(name) ?
+           this.urlbarPrefs.getIntPref(name) :
+           20;
+  },
```

Comment:
Yeah, the correct answer is definitely to use a preferences observer, but I think #58 is pretty far down the priority list. I figured that respecting the preference should, in theory, take precedent over any trivial speed gains, but if we're only going to have a few thousand users it's just not a big deal.